bindings/pyroot/pythonizations/src/TTreePyz.cxx

```diff
@@ -227,10 +227,12 @@ PyObject *PyROOT::SetBranchAddressPyz(PyObject * /* self */, PyObject *args)
 
       void *buf = 0;
       if (CPPInstance_Check(address)) {
+         ((CPPInstance *)address)->GetDatamemberCache(); // force creation of cache
```

Comment:
Does this have any performance implications for branches without array data members?

Reply:
One implication is that getting the internal C++ object of a Python proxy requires one extra access, since the pointer to the object becomes a pointer to a structure with more data (see [here](https://github.com/root-project/root/blob/master/bindings/pyroot/cppyy/CPyCppyy/src/CPPInstance.cxx#L73)). For what concerns the access to other (non-array) data members, it does not seem to have an impact, from what I can see [here](https://github.com/root-project/root/blob/master/bindings/pyroot/cppyy/CPyCppyy/src/CPPDataMember.cxx#L27).